core/domain/activity_jobs_one_off.py

```diff
@@ -471,3 +478,146 @@ def map(snapshot_model):
+                subtopic_models.SubtopicPageCommitLogEntryModel),
+            'id_string_format': 'subtopicpage-%s-%s'
+        },
+        'TopicRightsSnapshotMetadataModel': {
+            'parent_model_class': topic_models.TopicRightsModel,
+            'commit_log_model_class': topic_models.TopicCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s'
+        },
+        'CollectionRightsSnapshotMetadataModel': {
+            'parent_model_class': collection_models.CollectionRightsModel,
+            'commit_log_model_class': (
+                collection_models.CollectionCommitLogEntryModel),
+            'id_string_format': 'rights-%s-%s'
+        },
+        'ExplorationRightsSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationRightsModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s'
+        },
+    }
+    # This list consists of the rights snapshot metadata models for which
+    # the commit log model is not created when the commit cmd is "create"
+    # or "delete".
+    MODEL_NAMES_WITH_PARTIAL_COMMIT_LOGS = [
+        'CollectionRightsSnapshotMetadataModel',
+        'ExplorationRightsSnapshotMetadataModel'
+    ]
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of SnapshotMetadata models that is associated
+        with a CommitLogEntry model.
+        """
+        return ValidateSnapshotMetadataModelsJob.SNAPSHOT_METADATA_MODELS
+
+    @staticmethod
+    def map(snapshot_model):
+        """Implements the map function for this job."""
+        job_class = ValidateSnapshotMetadataModelsJob
+        class_name = snapshot_model.__class__.__name__
+        missing_commit_log_msg = (
+            '%s - MISSING COMMIT LOGS' % job_class.FAILURE_PREFIX)
+        found_commit_log_msg = 'FOUND COMMIT LOGS'
+
+        # Note: The subtopic snapshot ID is in the format
+        # '<topicId>-<subtopicNum>-<version>'.
+        model_id, version = snapshot_model.id.rsplit('-', 1)
+        model_properties = job_class.MODEL_NAMES_TO_PROPERTIES[class_name]
+        commit_log_id = (
+            model_properties['id_string_format'] % (model_id, version))
+        parent_model_class = (
+            model_properties['parent_model_class'].get_by_id(model_id))
+        commit_log_model_class = (
+            model_properties['commit_log_model_class'].get_by_id(
+                commit_log_id))
+        if class_name in job_class.MODEL_NAMES_WITH_PARTIAL_COMMIT_LOGS:
+            if snapshot_model.commit_type in ['create', 'delete']:
+                missing_commit_log_msg = (
+                    'COMMIT LOGS SHOULD NOT EXIST AND DOESNOT EXIST')
+                found_commit_log_msg = (
+                    '%s - COMMIT LOGS SHOULD NOT EXIST BUT EXISTS' % (
+                        job_class.FAILURE_PREFIX))
+
+        if commit_log_model_class is None:
```

Comment:
You can simplify this to reduce the branching:


```
message_prefix = (
    missing_commit_log_msg if commit_log_model_class is None else found_commit_log_msg)
yield ('%s - %s' % (message_prefix, class_name), snapshot_model.id)
```

Reply:
Done.